Lib/test/test_typing.py

```diff
@@ -3773,6 +3774,37 @@ def blah():
 
         blah()
 
+    def test_get_variants(self):
```

Comment:
How about naming this method `test_runtime_introspectable`, or something like that? I feel like we're testing the behaviour of `@overload` in this test case rather than the behaviour of `get_variants` (which, from the perspective of `@overload`, is an implementation detail). Would be good to make that clear. (The same goes for `test_get_variants_repeated` below.)

Reply:
Renaming it to test_variant_registry since it's testing the interaction with the variant registry